completion_aggregator/serializers.py

```diff
@@ -282,6 +282,65 @@ def get_mean(self, obj):
         return mean
 
 
+class CourseCompletionStatsSerializer(serializers.Serializer):
+    """
+    Serialize completions stats at the course level.
+    """
+
+    course_key = serializers.CharField()
+    completion = _CompletionSerializer(source='*')
+    username = serializers.SerializerMethodField()
+    mean = serializers.SerializerMethodField()
+
+    optional_fields = {'mean', 'username'}
+
+    def __init__(self, instance, requested_fields=frozenset(), *args, **kwargs):
+        """
+        Initialize a course completion stats serializer.
+
+        Add any requested optional fields.
+        """
+        super(CourseCompletionStatsSerializer, self).__init__(
+            instance, *args, **kwargs)
+        for field in self.optional_fields - requested_fields:
+            del self.fields[field]
+
+    def get_username(self, obj):
+        """
+        Serialize the username.
+        """
+        return obj.user.username
```

Comment:
I don't think there's any need for a username on this serializer.  In fact, the username won't be available, because it's just fetching aggregate results.